#! /bin/sh /usr/share/dpatch/dpatch-run
## 91_timeline-1.0.141-1.5.0.dpatch by Thomas Günther <tom@toms-cafe.de>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Changes for VDR >= 1.5.0 (applicable to VDR >= 1.4.5).

@DPATCH@
--- timeline-1.0.141/checkerOsd.c
+++ timeline-1.0.141/checkerOsd.c
@@ -339,8 +339,14 @@
 							{ // verschiedene Kanäle, genug Devices?
 								bool nofree=true;
 								
+#if APIVERSNUM >= 10500
+								cDevice *cd1=cDevice::GetDevice(ch1,-1,false);
+								cDevice *cd2=cDevice::GetDevice(ch2,-1,false);
+#else
 								cDevice *cd1=cDevice::GetDevice(ch1);
-								cDevice *cd2=cDevice::GetDevice(ch2),*cdt;
+								cDevice *cd2=cDevice::GetDevice(ch2);
+#endif
+								cDevice *cdt;
 								if (cd1->DeviceNumber()!=cd2->DeviceNumber())
 								{ // Sonderfall: ch2 kann gar nicht auf Device von ch1 empfangen werden
 									nofree=false;
